src/synchronizer.ts

```diff
@@ -112,11 +113,19 @@ export async function createSynchronizer(
     const processedSnapshots = createProcessedSnapshotsComponent(components)
 
     logger.info('Starting to deploy entities from snapshots.')
-    const deploymentsFromSnapshots: (() => Promise<void>)[] = []
+
+    const deploymentsProcessorsQueue = new PQueue({
+      concurrency: 10,
+      autoStart: true
+    })
 
     for (const snapshotInfo of snapshotInfoByHash.values()) {
-      deploymentsFromSnapshots.push(async () => {
-        const stream = getDeployedEntitiesStreamFromSnapshot({ ...components, processedSnapshots }, options, snapshotInfo)
+      await deploymentsProcessorsQueue.add(async () => {
```

Comment:
That `await` will prevent the concurrency
```suggestion
     void deploymentsProcessorsQueue.add(async () => {
```

Reply:
Thanks! Nice catch!